### DISCUSSION TITLE
Map resets every cleanup

### DISCUSSION THREAD

**Discussion by Drako12** [Hypfer/Valetudo#1731] [Q&A / Support]
I have a Xiaomi Mijia STYJ02YM, I flashed with a V6 version and I had no issues flashing or installing valetudo.

After adding the wifi and going into the main valetudo page for the first time I got that message  where it shows a button to do a full cleanup to build the map, I did that, the robot started cleaning as expected and it did until it ran out of battery, but the map wasnt complete yet. The robot docked, I waited until the battery was full and hit to clean again, but all it did was clean the map and start again, doing the exact same rote (to create the map) and again, without finishing the house and running out of battery, and everytime I reload the main valetudo page I get the "you are using for the first time" message.

I clicked the "persistant map" switch on/off and deleted the map and started again, but same thing happened.

How can I have a map of my full house saved? thanks!

**Comment by Hypfer** [comment]
When starting the full cleanup, the confirmation dialog contains the following message:

> The robot needs to return to the dock on its own to save the newly created map.
> Do not interfere with the cleanup or else it won't be saved.

At least to my knowledge, the robot should drive back to the dock when it reaches low battery levels, charge backup and then continue on its own where it left off once it is fully charged. Interrupting that by pressing any buttons will start a new cleanup, voiding the progress.

Make sure that "persistent map" is enabled before starting the initial full cleanup. You can also help the robot clean a larger area by reducing the fan speed to the minimum as that will consume less battery.

> everytime I reload the main valetudo page I get the "you are using for the first time" message.

The "Hide" button hides the welcome dialog until the next refresh. The "Do not show again" button hides it permanently.

**Comment by Drako12** [comment]
I thought the robot would start again after recharging and continue, but It didn't,  It went into full charge and did nothing else.

When it ran out of battery and went into docking, the robot was in the "cleaning" state while charging, but around 70-80% it just stopped and went into "docked" state. Also, the segments weren't marked, the map was full blue with the path the robot did while cleaning. So when I hit the play button again, the map was deleted and it started a new one.

Right now I think I managed a workaround, waiting for the full charge to test. Right after it ran out of battery and went into docking I clicked the pause button and the map did the segments (maps has regions with colors). 

Im just not sure if I had to hit the play button or I have to always clean using segments, if I hit the play button the map will disappear again?

**Comment by Drako12** [comment]
didnt really work..

I can't get the map to complete properly because everytime it runs it will delete the segments/places that it didnt manage to go before running out of battery.

I will try again closing the doors from the places where it already went around but didn't clean yet.

Blue areas were segments before this last cleanup, but I closed the doors on purpose so it would manage to get to the other places before running out of battery. After it went back into docking the segments disappeard and the area turned blue.

![brave_2023-03-05_11-57-10](https://user-images.githubusercontent.com/768180/222968176-91c98002-01f5-417d-adbe-a51673e7c46d.png)

**Reply by iherevn** [reply]
I think your house is too big to clean and map at the same time. Please run fast mapping only mode.
